### ISSUE TITLE
deadlink not reported with `file-path`

### ISSUE THREAD

**Issue by DanielYang59** [tcort/github-action-markdown-link-check#8]
With the [following workflow](https://github.com/FAIRmat-NFDI/nomad-docs/pull/283/changes) (debug purpose):
```yml
      - name: Check links in markdown files
        uses: tcort/github-action-markdown-link-check@v1
        with:
          config-file: .github/workflows/configs/markdown-link-check.json
          file-path: "./README.md"

      - uses: actions/setup-node@v6
        with:
          node-version: "20"
      - name: Probe problematic file only
        run: npx -y markdown-link-check@3.14.2 ./README.md -c .github/workflows/configs/markdown-link-check.json -v
```

And README.md contains a dead link:
```
https://gaskjdhaskhdaksjhdjahsdaksjhdithub.com/FAIRmat-NFDI/nomad-docs/issues/new
```

First passed:
```
Using markdown-link-check configuration file: .github/workflows/configs/markdown-link-check.json
USE_QUIET_MODE: no
USE_VERBOSE_MODE: no
FOLDER_PATH: .
MAX_DEPTH: -1
CHECK_MODIFIED_FILES: no
FILE_EXTENSION: .md
FILE_PATH: ./README.md
+ find . -name '*.md' -not -path './node_modules/*' -exec markdown-link-check '{}' --config .github/workflows/configs/markdown-link-check.json ';'
+ set +x
+ find . -type f '(' -wholename ./README.md ')' -not -path './node_modules/*' -exec markdown-link-check '{}' --config .github/workflows/configs/markdown-link-check.json ';'
+ set +x
=========================> MARKDOWN LINK CHECK <=========================

[✔] All links are good!

=========================================================================
```

And second failed (expected behaviour):
```
FILE: ./README.md
  [✖] https://gaskjdhaskhdaksjhdjahsdaksjhdithub.com/FAIRmat-NFDI/nomad-docs/issues/new → Status: 0 Error: getaddrinfo ENOTFOUND gaskjdhaskhdaksjhdjahsdaksjhdithub.com
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:122:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'gaskjdhaskhdaksjhdjahsdaksjhdithub.com'
}
  [✓] #external-contribution-instructions → Status: 200
  [✓] https://github.com/FAIRmat-NFDI/nomad-docs/blob/main/docs/writing_guide.md → Status: 200
  [✓] https://github.com/FAIRmat-NFDI/nomad-distro-dev?tab=readme-ov-file#day-to-day-development → Status: 200
  [✓] #help-to-install-python-311-or-312 → Status: 200
  [✓] https://nomad-lab.eu/prod/v1/develop/docs/index.html → Status: 200
  [✓] https://www.python.org/downloads/release/python-3110/ → Status: 200
  7 links checked.
  ERROR: 1 dead link found!
  [✖] https://gaskjdhaskhdaksjhdjahsdaksjhdithub.com/FAIRmat-NFDI/nomad-docs/issues/new → Status: 0
Error: Process completed with exit code 1.
```
